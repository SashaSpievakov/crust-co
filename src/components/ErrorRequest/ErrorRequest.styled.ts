import styled from 'styled-components';

import { Container } from '@src/styles/Base.styled';

const Article = styled(Container)`
  margin: 100px auto;
  text-align: center;
`;

const ErrorHeading = styled.h2`
  font-size: 45px;
  font-weight: 700;
  line-height: 1.2;
  margin-bottom: 5px;

  @media screen and (max-width: 550px) {
    font-size: 30px;
    padding: 0 15px;
    margin-bottom: 0;
  }

  @media screen and (max-width: 450px) {
    font-size: 26px;
  }
`;

export { Article, ErrorHeading };
