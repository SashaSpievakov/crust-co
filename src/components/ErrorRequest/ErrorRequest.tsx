import { ErrorParagraph } from 'src/styles/Base.styled';
import { Article, ErrorHeading } from './ErrorRequest.styled';

const ErrorRequest = () => {
  return (
    <Article data-testid="itemPage">
      <ErrorHeading>Error: failed to get data from the server</ErrorHeading>
      <ErrorParagraph>
        We are already working on that. Try again later
      </ErrorParagraph>
    </Article>
  );
};

export default ErrorRequest;
