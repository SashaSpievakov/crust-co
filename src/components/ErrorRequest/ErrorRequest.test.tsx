import '@testing-library/jest-dom';

import { rendererWithProviders } from '../../tests/helpers';
import ErrorRequest from './ErrorRequest';

describe('ErrorRequest Tests', () => {
  test('renders the ErrorRequest UI component', () => {
    const snapshot = rendererWithProviders(<ErrorRequest />);
    expect(snapshot).toMatchSnapshot();
  });
});
