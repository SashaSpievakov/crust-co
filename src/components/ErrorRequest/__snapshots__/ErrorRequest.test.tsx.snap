// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorRequest Tests renders the ErrorRequest UI component 1`] = `
Object {
  "children": Array [
    <h2
      className="sc-GhhNo hhgdai"
    >
      Error: failed to get data from the server
    </h2>,
    <p
      className="sc-idXgbr daODfB"
    >
      We are already working on that. Try again later
    </p>,
  ],
  "props": Object {
    "className": "sc-bjfHbI sc-fbYMXx cXrwfW cALrQe",
    "data-testid": "itemPage",
  },
  "type": "article",
}
`;
